subsys/net/lib/lwm2m/lwm2m_engine.c

```diff
@@ -4628,28 +4534,129 @@ void lwm2m_socket_del(struct lwm2m_ctx *ctx)
 	}
 }
 
-/* LwM2M main work loop */
+static void create_notifications(sys_slist_t *observer_list,
+				 const int64_t timestamp)
+{
+	struct observe_node *obs;
+	int rc;
+
+	SYS_SLIST_FOR_EACH_CONTAINER(observer_list, obs, node) {
+		if (obs->event_timestamp > obs->last_timestamp &&
+		    timestamp > obs->last_timestamp +
+		    MSEC_PER_SEC * obs->min_period_sec) {
+			rc = generate_notify_message(obs, true);
+			if (rc == -ENOMEM) {
+				/* no memory/messages available, retry later */
+				return;
+			}
+			obs->last_timestamp = timestamp;
+		} else if (timestamp >
```

Comment:
This does not match the original condition:
```
		} else if (obs->max_period_sec > 0 &&
			   timestamp > obs->last_timestamp +
				MSEC_PER_SEC * obs->max_period_sec) {
```
Note the `obs->max_period_sec > 0` check missing.

Reply:
Also a carry over from the earlier version this was originally based on, good catch :+1: